tests/e2e_tests/config.py

```diff
@@ -0,0 +1,111 @@
+import base64
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import Literal, Protocol
+
+import boto3
+from playwright.sync_api import HttpCredentials
+
+
+class EndToEndTestSecrets(Protocol):
+    @property
+    def HTTP_BASIC_AUTH(self) -> HttpCredentials | None: ...
+
+    @property
+    def JWT_SIGNING_KEY(self) -> str: ...
+
+    @property
+    def NOTIFY_FIND_API_KEY(self) -> str: ...
+
+    @property
+    def NOTIFY_SUBMIT_API_KEY(self) -> str: ...
+
+
+class LocalEndToEndSecrets:
+    @property
+    def HTTP_BASIC_AUTH(self) -> None:
+        return None
+
+    @property
+    def JWT_SIGNING_KEY(self) -> str:
+        _test_private_key_path = str(Path(__file__).parent.parent) + "/keys/rsa256/private.pem"
+        with open(_test_private_key_path, mode="r") as private_key_file:
+            rsa256_private_key = private_key_file.read()
+
+        return rsa256_private_key
+
+    @property
+    def NOTIFY_FIND_API_KEY(self) -> str:
+        return os.environ["E2E_NOTIFY_FIND_API_KEY"]
+
+    @property
+    def NOTIFY_SUBMIT_API_KEY(self) -> str:
+        return os.environ["E2E_NOTIFY_SUBMIT_API_KEY"]
+
+
+class AWSEndToEndSecrets:
+    def __init__(self, e2e_env: Literal["dev", "test"], e2e_aws_vault_profile: str | None):
+        self.e2e_env = e2e_env
+        self.e2e_aws_vault_profile = e2e_aws_vault_profile
+
+        if self.e2e_env == "prod":  # type: ignore[comparison-overlap]
+            raise ValueError("shouldn't be possible, but also must never happen")
```

Comment:
I'm doing this intentionally to help prevent bad things happening.

I specifically do not want someone to look at the `Literal['dev', 'test']` and think "oh hey I can just add prod and away we go". I do not want that to happen - because that would mean reading prod secrets, and no - I want to add layers of "stop and think what you're doing" before anyone does that.